### PR TITLE
Runtime core runtime status

### PR DESCRIPTION
**Pull request by junmendoza**
@ke-yu @aparajit-pratap 

1. This is change involves moving the RuntimeStatus instance from Core to RuntimeCore
2. RuntimeStatus is only used for logging info at runtime and it has nothing to do with compilation

### CODE REVIEW

File: src/Engine/ProtoCore/DSASM/Executive.cs [hunk 1]
```diff
@@ -807,7 +807,7 @@ private bool GetSymbolIndexedIntoList(string symbol, out List<int> indexList)
             }
 
             // Get the cached callsite, creates a new one for a first-time call
-            CallSite callsite = exe.RuntimeData.GetCallSite(exe.RuntimeData.ExecutingGraphnode, classIndex, fNode.name, exe, core.RunningBlock, runtimeCore.RuntimeOptions, core.RuntimeStatus);
+            CallSite callsite = exe.RuntimeData.GetCallSite(exe.RuntimeData.ExecutingGraphnode, classIndex, fNode.name, exe, core.RunningBlock, runtimeCore.RuntimeOptions, runtimeCore.RuntimeStatus);
             Validity.Assert(null != callsite);
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/DSASM/Executive.cs]: split - line too long

File: src/Engine/ProtoCore/DSASM/Stack.cs [hunk 2]
```diff
@@ -455,10 +455,11 @@ private static bool CompareStackValuesFromHeap(StackValue sv1, StackValue sv2, C
         // heaper method to support negative index into stack
         public static StackValue GetValue(this HeapElement hs, int ix, Core core)
         {
+            RuntimeCore runtimeCore = core.__TempCoreHostForRefactoring;
             int index = ix < 0 ? ix + hs.VisibleSize : ix;
             if (index >= hs.VisibleSize || index < 0)
             {
-                core.RuntimeStatus.LogWarning(ProtoCore.Runtime.WarningID.kOverIndexing, Resources.kArrayOverIndexed);
+                runtimeCore.RuntimeStatus.LogWarning(ProtoCore.Runtime.WarningID.kOverIndexing, Resources.kArrayOverIndexed);
                 return StackValue.Null;
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/DSASM/Stack.cs]: line too long

File: src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs [hunk 3]
```diff
@@ -506,7 +506,7 @@ private StackValue DotMethod(StackValue lhs, StackFrame stackFrame, DSASM.Execut
 
             if (!isValidThisPointer || (!thisObject.IsPointer && !thisObject.IsArray))
             {
-                core.RuntimeStatus.LogWarning(WarningID.kDereferencingNonPointer, Resources.kDeferencingNonPointer);
+                runtimeCore.RuntimeStatus.LogWarning(WarningID.kDereferencingNonPointer, Resources.kDeferencingNonPointer);
                 return StackValue.Null;
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs]: one more

File: src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs [hunk 4]
```diff
@@ -631,7 +631,7 @@ private StackValue DotMethod(StackValue lhs, StackFrame stackFrame, DSASM.Execut
                                                null);
 
             ProtoCore.CallSite callsite = runtimeData.GetCallSite(
-                runtimeData.ExecutingGraphnode, thisObjectType, functionName, runtime.exe, core.RunningBlock, core.Options, core.RuntimeStatus);
+                runtimeData.ExecutingGraphnode, thisObjectType, functionName, runtime.exe, core.RunningBlock, core.Options, runtimeCore.RuntimeStatus);
             Validity.Assert(null != callsite);
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs]: yet another

File: src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs [hunk 5]
```diff
@@ -1262,10 +1265,11 @@ internal static StackValue Difference(StackValue sv1, StackValue sv2, ProtoCore.
         //Union
         internal static StackValue Union(StackValue sv1, StackValue sv2, ProtoCore.DSASM.Interpreter runtime, ProtoCore.Runtime.Context context)
         {
+            RuntimeCore runtimeCore = runtime.runtime.RuntimeCore;
             if ((Rank(sv1, runtime) != 1) || (Rank(sv2, runtime) != 1))
             {
                 //LC urgent patch
-                runtime.runtime.Core.RuntimeStatus.LogWarning(ProtoCore.Runtime.WarningID.kTypeMismatch, "Both arguments were expected to be one-dimensional array type!");
+                runtimeCore.RuntimeStatus.LogWarning(ProtoCore.Runtime.WarningID.kTypeMismatch, "Both arguments were expected to be one-dimensional array type!");
                 return DSASM.StackValue.Null;
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs]: split

File: src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs [hunk 6]
```diff
@@ -1700,10 +1712,11 @@ internal static bool Contains(StackValue sv1, StackValue sv2, ProtoCore.DSASM.In
         }
         internal static bool ContainsArray(StackValue sv1, StackValue sv2, ProtoCore.DSASM.Interpreter runtime)
         {
+            RuntimeCore runtimeCore = runtime.runtime.RuntimeCore;
             if (!sv1.IsArray)
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs]: just use `var`

File: src/Engine/ProtoCore/RuntimeCore.cs [hunk 7]
```diff
@@ -40,9 +44,16 @@ public RuntimeCore(Options runtimeOptions, Executable executable, ProtoCore.Runt
 
         public Executable DSExecutable { get; private set; }
         public Options RuntimeOptions { get; private set; }
+        public RuntimeStatus RuntimeStatus { get; set; }
 
         public RuntimeMemory RuntimeMemory { get; set; }
-        private ProtoCore.Runtime.Context context;
+        public ProtoCore.Runtime.Context context { get; set; }
+
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/RuntimeCore.cs]: make property name start with caps

File: src/Engine/ProtoCore/RuntimeStatus.cs [hunk 8]
```diff
@@ -51,20 +51,28 @@ public struct WarningEntry
...
+        {
+            get;
+            set;
+        }
+
+        public IOutputStream WebMessageHandler
+        {
+            get;
+            set;
         }
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/RuntimeStatus.cs]: `public IOutputStream WebMessageHandler { get; set; }` looks better

File: src/Engine/ProtoCore/RuntimeStatus.cs [hunk 9]
```diff
@@ -51,20 +51,28 @@ public struct WarningEntry
     public class RuntimeStatus
     {
         private ProtoCore.Core core;
+        private ProtoCore.RuntimeCore runtimeCore;
```
Comment (aparajit-pratap) [inline src/Engine/ProtoCore/RuntimeStatus.cs]: Do you need a handle to the compile Core here?

**Comment by aparajit-pratap** [comment]
@junmendoza other than the above comments, it looks good to me.

File: src/Engine/ProtoCore/DSASM/Executive.cs (hunk 1 above)
Comment (junmendoza) [inline src/Engine/ProtoCore/DSASM/Executive.cs]: Done

File: src/Engine/ProtoCore/DSASM/Stack.cs (hunk 2 above)
Comment (junmendoza) [inline src/Engine/ProtoCore/DSASM/Stack.cs]: Done

File: src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs (hunk 4 above)
Comment (junmendoza) [inline src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs]: Done

File: src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs (hunk 5 above)
Comment (junmendoza) [inline src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs]: Done

File: src/Engine/ProtoCore/RuntimeCore.cs (hunk 7 above)
Comment (junmendoza) [inline src/Engine/ProtoCore/RuntimeCore.cs]: Done

File: src/Engine/ProtoCore/RuntimeStatus.cs (hunk 9 above)
Comment (junmendoza) [inline src/Engine/ProtoCore/RuntimeStatus.cs]: No, the Compilecore is not necessary once the complete swap is done

File: src/Engine/ProtoCore/RuntimeStatus.cs (hunk 8 above)
Comment (junmendoza) [inline src/Engine/ProtoCore/RuntimeStatus.cs]: Done

**Pull request merged**